haystack/nodes/retriever/dense.py

```diff
@@ -1898,3 +1898,303 @@ def save(self, save_dir: Union[Path, str]) -> None:
         :type save_dir: Union[Path, str]
         """
         self.embedding_encoder.save(save_dir=save_dir)
+
+
+class MultihopDenseRetriever(EmbeddingRetriever):
+    """
+    Retriever that applies iterative retrieval using a shared encoder for query and passage.
+    See original paper for more details:
+
+    Xiong, Wenhan, et. al. (2020): "Answering complex open-domain questions with multi-hop dense retrieval"
+    (https://arxiv.org/abs/2009.12756)
+    """
+
+    def __init__(
+        self,
+        document_store: BaseDocumentStore,
+        embedding_model: str,
+        model_version: Optional[str] = None,
+        num_iterations: int = 2,
+        use_gpu: bool = True,
+        batch_size: int = 32,
+        max_seq_len: int = 512,
+        model_format: str = "farm",
+        pooling_strategy: str = "reduce_mean",
+        emb_extraction_layer: int = -1,
+        top_k: int = 10,
+        progress_bar: bool = True,
+        devices: Optional[List[Union[str, torch.device]]] = None,
+        use_auth_token: Optional[Union[str, bool]] = None,
+        scale_score: bool = True,
+        embed_meta_fields: List[str] = [],
+    ):
+        """
+        Same parameters as `EmbeddingRetriever` except
```

Comment:
I'd like to see descriptions for all the parameters that can be set in the __init__ method. Feel free to just copy them from the `EmbeddingRetriever`.

Reply:
Done